src/node_modules.cc

```diff
@@ -0,0 +1,420 @@
+
+  auto cache_entry = binding_data->package_configs_.find(path);
+  if (cache_entry != binding_data->package_configs_.end()) {
+    return cache_entry->second;
+  }
+
+  std::string input;
+  if (ReadFileSync(&input, path.c_str()) < 0) {
+    return std::nullopt;
+  }
+  input.reserve(input.size() + simdjson::SIMDJSON_PADDING);
+
+  std::string_view json_string(input);
+  if (input.size() >= 3 && 0 == memcmp(input.data(), "\xEF\xBB\xBF", 3)) {
+    json_string.remove_prefix(3);  // Skip UTF-8 BOM.
+  }
+
+  simdjson::ondemand::document document;
+  simdjson::ondemand::object main_object;
+  simdjson::error_code error =
+      binding_data->json_parser.iterate(json_string, json_string.max_size())
+          .get(document);
+
+  const auto throw_invalid_package_config = [error_context, path, realm]() {
+    if (error_context == nullptr) {
+      THROW_ERR_INVALID_PACKAGE_CONFIG(
+          realm->isolate(), "Invalid package config %s.", path.data());
+    } else if (error_context->base.has_value()) {
+      auto file_url = ada::parse(error_context->base.value());
+      CHECK(file_url);
+      auto file_path = url::FileURLToPath(realm->env(), *file_url);
+      CHECK(file_path.has_value());
+      THROW_ERR_INVALID_PACKAGE_CONFIG(
+          realm->isolate(),
+          "Invalid package config %s while importing \"%s\" from %s.",
+          path.data(),
+          error_context->specifier.c_str(),
+          file_path->c_str());
+    } else {
+      THROW_ERR_INVALID_PACKAGE_CONFIG(
+          realm->isolate(), "Invalid package config %s.", path.data());
+    }
+
+    return std::nullopt;
+  };
+
+  if (error || document.get_object().get(main_object)) {
+    return throw_invalid_package_config();
+  }
+
+  simdjson::ondemand::raw_json_string key;
+  simdjson::ondemand::value value;
+  std::string_view field_value;
+  simdjson::ondemand::json_type field_type;
+  PackageConfig package_config{};
+
+  for (auto field : main_object) {
+    // Throw error if getting key or value fails.
+    if (field.key().get(key) || field.value().get(value)) {
+      return throw_invalid_package_config();
+    }
+
+    if (key == "name") {
+      value.get_string(package_config.name);
+    } else if (key == "main") {
+      value.get_string(package_config.main);
+    } else if (key == "exports") {
+      if (value.type().get(field_type)) {
+        return throw_invalid_package_config();
+      }
+      switch (field_type) {
+        case simdjson::ondemand::json_type::object:
+        case simdjson::ondemand::json_type::array: {
+          if (value.raw_json().get(field_value)) {
+            return throw_invalid_package_config();
+          }
+          package_config.exports = std::string(field_value);
+          break;
+        }
+        case simdjson::ondemand::json_type::string: {
+          if (value.get_string().get(field_value)) {
+            return throw_invalid_package_config();
+          }
+          package_config.exports = std::string(field_value);
```

Comment:
It might be possible to do something like...
```C++
if(value.get_string(package_config.exports)) {
  return throw_invalid_package_config();
}
```

Thus saving a line of code.

Reply:
I love it! Thanks for the review.